pkg/api/v1alpha1/cluster_types.go

```diff
@@ -323,9 +325,102 @@ func (n *ClusterNetwork) Equal(o *ClusterNetwork) bool {
 	if n == nil || o == nil {
 		return false
 	}
-	return SliceEqual(n.Pods.CidrBlocks, o.Pods.CidrBlocks) &&
-		SliceEqual(n.Services.CidrBlocks, o.Services.CidrBlocks) &&
-		n.CNI == o.CNI && n.DNS.ResolvConf.Equal(o.DNS.ResolvConf)
+
+	if !CNIPluginSame(*n, *o) {
+		return false
+	}
+	if o.CNIConfig != nil {
```

Comment:
nit: we can remove this, `CNIConfig.Equal` handles this case

Reply:
this is needed in case the older cluster spec does not have cniConfig set, and is using the deprecated field